internal/server/pool_test.go

```diff
@@ -53,7 +54,455 @@ func TestPoolSpike(t *testing.T) {
+		cfg.MaxIdleTime = cfg.MaxLifetime * 10
+
+		rt := be.Relaxed(t)
+
+		p, err := NewPool(cfg)
+		be.NilErr(rt, err)
+		defer func() { be.NilErr(rt, p.Close()) }()
+
+		conn0, err := p.AcquireConn()
+		be.NilErr(rt, err)
+
+		conn1, err := p.AcquireConn()
+		be.NilErr(rt, err)
+
+		be.Equal(t, 2, p.Stats().OpenConnections)
+		be.Equal(t, 0, p.Stats().Idle)
+
+		p.ReleaseConn(conn0)
+		p.ReleaseConn(conn1)
+
+		be.Equal(t, 2, p.Stats().OpenConnections)
+		be.Equal(t, 2, p.Stats().Idle)
+
+		time.Sleep(cfg.MaxLifetime + time.Second)
+
+		be.Equal(t, 0, p.Stats().OpenConnections)
+		be.Equal(t, 0, p.Stats().Idle)
+		be.Equal(t, 2, p.Stats().MaxLifetimeClosed)
+	})
+
+	t.Run("release conn", func(t *testing.T) {
+		cfg := genBasePoolConfig()
+		cfg.MaxOpen = 3
+		cfg.MaxIdle = 3
+		cfg.MaxLifetime = 1 * time.Second
+		cfg.MaxIdleTime = cfg.MaxLifetime * 10
+
+		rt := be.Relaxed(t)
+
+		p, err := NewPool(cfg)
+		be.NilErr(rt, err)
+		defer func() { be.NilErr(rt, p.Close()) }()
+
+		conn0, err := p.AcquireConn()
+		be.NilErr(rt, err)
+
+		conn1, err := p.AcquireConn()
+		be.NilErr(rt, err)
+
+		be.Equal(t, 2, p.Stats().OpenConnections)
+		be.Equal(t, 0, p.Stats().Idle)
+
+		time.Sleep(cfg.MaxLifetime + time.Second)
+
+		p.ReleaseConn(conn0)
+		p.ReleaseConn(conn1)
+
+		be.Equal(t, 0, p.Stats().OpenConnections)
+		be.Equal(t, 0, p.Stats().Idle)
+		be.Equal(t, 2, p.Stats().MaxLifetimeClosed)
+	})
+}
+
+func TestMaxLifetimeUpdate(t *testing.T) {
+	cfg := genBasePoolConfig()
+	cfg.MaxOpen = 3
+	cfg.MaxIdle = 3
+	cfg.MaxLifetime = 1 * time.Second
+	cfg.MaxIdleTime = cfg.MaxLifetime * 10
+
+	rt := be.Relaxed(t)
+
+	p, err := NewPool(cfg)
+	be.NilErr(rt, err)
+	defer func() { be.NilErr(rt, p.Close()) }()
+
+	conn0, err := p.AcquireConn()
+	be.NilErr(rt, err)
+
+	conn1, err := p.AcquireConn()
+	be.NilErr(rt, err)
+
+	be.Equal(t, 2, p.Stats().OpenConnections)
+	be.Equal(t, 0, p.Stats().Idle)
+
+	p.ReleaseConn(conn0)
+	p.ReleaseConn(conn1)
+
+	be.Equal(t, 2, p.Stats().OpenConnections)
+	be.Equal(t, 2, p.Stats().Idle)
+
+	time.Sleep(100 * time.Millisecond)
+
+	be.Equal(t, 2, p.Stats().OpenConnections)
+	be.Equal(t, 2, p.Stats().Idle)
+
+	p.SetConnMaxLifetime(100 * time.Millisecond)
+
+	time.Sleep(p.maxLifetime + time.Second)
+
+	be.Equal(t, 0, p.Stats().OpenConnections)
+	be.Equal(t, 0, p.Stats().Idle)
+	be.Equal(t, 2, p.Stats().MaxLifetimeClosed)
+}
+
+func TestMaxOpenConnBlock(t *testing.T) {
+	cfg := genBasePoolConfig()
+	cfg.MaxOpen = 2
+
+	rt := be.Relaxed(t)
+
+	p, err := NewPool(cfg)
+	be.NilErr(rt, err)
+	defer func() { be.NilErr(rt, p.Close()) }()
+
+	conn0, err := p.AcquireConn()
+	be.NilErr(rt, err)
+
+	conn1, err := p.AcquireConn()
+	be.NilErr(rt, err)
+
+	var wg sync.WaitGroup
+	connChan := make(chan *ServerConn, 1)
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		sc, err2 := p.AcquireConn()
+		be.NilErr(rt, err2)
+		connChan <- sc
+	}()
+
+	// Wait to let some goroutines wait
+	// and ensure that none can acquire without releasing.
```

Comment:
This should refer to the goroutine, in singular, right? Or am I misunderstanding something?
```suggestion
	// Sleep to let the goroutine wait
	// and ensure that it cannot acquire without releasing.
```

Reply:
You are right.